From e2f95e23fde16bf99079f3f3257edd1e9fa04b75 Mon Sep 17 00:00:00 2001
Message-ID: <e2f95e23fde16bf99079f3f3257edd1e9fa04b75.1699403993.git.sharlatanus@gmail.com>
In-Reply-To: <0117df90b1ef43a67d4fa86e5b58e60f7c0a52e5.1699403992.git.sharlatanus@gmail.com>
References: <0117df90b1ef43a67d4fa86e5b58e60f7c0a52e5.1699403992.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 29 Oct 2023 23:50:23 +0000
Subject: [PATCH 64/67] gnu: go-github-com-getsentry-raven-go: Move to (gnu
 packages golang-web).

* gnu/packages/golang.scm (go-github-com-getsentry-raven-go): Move from
here...
* gnu/packages/golang-web.scm: ...to here.

Change-Id: If7e155b3e85d8bda8fd099f6a4d1eb6b87e9eee3
---
 gnu/packages/golang-web.scm | 31 +++++++++++++++++++++++++++++++
 gnu/packages/golang.scm     | 27 ---------------------------
 2 files changed, 31 insertions(+), 27 deletions(-)

diff --git a/gnu/packages/golang-web.scm b/gnu/packages/golang-web.scm
index c046e0d46a..41c8c37a0b 100644
--- a/gnu/packages/golang-web.scm
+++ b/gnu/packages/golang-web.scm
@@ -9,6 +9,7 @@
 ;;; Copyright © 2020 raingloom <raingloom@riseup.net>
 ;;; Copyright © 2020-2022 Efraim Flashner <efraim@flashner.co.il>
 ;;; Copyright © 2021 Collin J. Doering <collin@rekahsoft.ca>
+;;; Copyright © 2021 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2021 Philip McGrath <philip@philipmcgrath.com>
 ;;; Copyright © 2021 Raghav Gururajan <rg@raghavgururajan.name>
 ;;; Copyright © 2021 Sarah Morgensen <iskarian@mgsn.dev>
@@ -45,6 +46,7 @@ (define-module (gnu packages golang-web)
   #:use-module (gnu packages)
   #:use-module (gnu packages golang)
   #:use-module (gnu packages golang-check)
+  #:use-module (gnu packages tls)
   #:use-module (gnu packages web))
 
 ;;; Commentary:
@@ -521,6 +523,35 @@ (define-public go-github-com-francoispqt-gojay
     (home-page "https://github.com/francoispqt/gojay")
     (license license:expat)))
 
+;; TODO: This repository has been archived by the owner on Aug 30, 2023. It is
+;; now read-only. The raven-go SDK is no longer maintained and was superseded
+;; by the sentry-go
+(define-public go-github-com-getsentry-raven-go
+  (let ((commit "5c24d5110e0e198d9ae16f1f3465366085001d92")
+        (revision "0"))
+    (package
+      (name "go-github-com-getsentry-raven-go")
+      (version (git-version "0.2.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/getsentry/raven-go")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "0lvc376sq8r8jhy2v1m6rf1wyld61pvbk0x6j9xpg56ivqy69xs7"))))
+      (build-system go-build-system)
+      (arguments
+       '(#:import-path "github.com/getsentry/raven-go"))
+      (propagated-inputs
+       (list go-github-com-certifi-gocertifi go-github-com-pkg-errors))
+      (home-page "https://github.com/getsentry/raven-go")
+      (synopsis "Sentry client in Go")
+      (description "This package is a Go client API for the Sentry event/error
+logging system.")
+      (license license:bsd-3))))
+
 (define-public go-github-com-go-chi-chi-v5
   (package
     (name "go-github-com-go-chi-chi-v5")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index d53fa5864a..d4d8d352a5 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -4002,33 +4002,6 @@ (define-public go-github-com-burntsushi-toml
 is similar to Go's standard library @code{json} and @code{xml} package.")
     (license license:expat)))
 
-(define-public go-github-com-getsentry-raven-go
-  (let ((commit "5c24d5110e0e198d9ae16f1f3465366085001d92")
-        (revision "0"))
-    (package
-      (name "go-github-com-getsentry-raven-go")
-      (version (git-version "0.2.0" revision commit))
-      (source
-       (origin
-         (method git-fetch)
-         (uri (git-reference
-               (url "https://github.com/getsentry/raven-go")
-               (commit commit)))
-         (file-name (git-file-name name version))
-         (sha256
-          (base32
-           "0lvc376sq8r8jhy2v1m6rf1wyld61pvbk0x6j9xpg56ivqy69xs7"))))
-      (build-system go-build-system)
-      (arguments
-       '(#:import-path "github.com/getsentry/raven-go"))
-      (propagated-inputs
-       (list go-github-com-certifi-gocertifi go-github-com-pkg-errors))
-      (home-page "https://github.com/getsentry/raven-go")
-      (synopsis "Sentry client in Go")
-      (description "This package is a Go client API for the Sentry event/error
-logging system.")
-      (license license:bsd-3))))
-
 (define-public go-github-com-hashicorp-go-uuid
   (package
     (name "go-github-com-hashicorp-go-uuid")
-- 
2.41.0

